aspnetcore/blazor/common/samples/3.x/BlazorServerEFCoreSample/BlazorServerDbContextExample/Pages/Index.razor

```diff
@@ -0,0 +1,242 @@
+        <div class="row">
+            <div class="col-6">
+                Page @Filters.PageHelper.Page of @Filters.PageHelper.PageCount:
+                showing @Filters.PageHelper.PageItems of
+                @Filters.PageHelper.TotalItemCount items.
+                <a disabled="@(Filters.Loading || !Filters.PageHelper.HasPrev)"
+                   class="btn btn-primary @IsDisabled(Filters.PageHelper.HasPrev)"
+                   href="/@Filters.PageHelper.PrevPage">
+                    Previous
+                </a>
+                <a disabled="@(Filters.Loading || !Filters.PageHelper.HasNext)"
+                   class="btn btn-primary @IsDisabled(Filters.PageHelper.HasNext)"
+                   href="/@Filters.PageHelper.NextPage">
+                    Next
+                </a>
+            </div>
+        </div>
+        <div class="row">&nbsp;</div>
+        <div class="row contact-header">
+            <div class="col-1">&nbsp;</div>
+            <div class="col-2"
+                 @onclick="@(async ()=>await ToggleAsync(ContactFilterColumns.Name))">
+                <SortIndicator Column="@(ContactFilterColumns.Name)" />&nbsp;Name
+            </div>
+            <div class="col-2"
+                 @onclick="@(async ()=>await ToggleAsync(ContactFilterColumns.Phone))">
+                <SortIndicator Column="@(ContactFilterColumns.Phone)" />&nbsp;📞 Phone
+            </div>
+            <div class="col-3"
+                 @onclick="@(async ()=>await ToggleAsync(ContactFilterColumns.Street))">
+                <SortIndicator Column="@(ContactFilterColumns.Street)" />&nbsp;🏠 Street Address
+            </div>
+            <div class="col-2"
+                 @onclick="@(async ()=>await ToggleAsync(ContactFilterColumns.City))">
+                <SortIndicator Column="@(ContactFilterColumns.City)" />&nbsp;City
+            </div>
+            <div class="col-1"
+                 @onclick="@(async ()=>await ToggleAsync(ContactFilterColumns.State))">
+                <SortIndicator Column="@(ContactFilterColumns.State)" />&nbsp;State
+            </div>
+            <div class="col-1"
+                 @onclick="@(async ()=>await ToggleAsync(ContactFilterColumns.ZipCode))">
+                <SortIndicator Column="@(ContactFilterColumns.ZipCode)" />&nbsp;ZipCode
+            </div>
+        </div>
+        @if (Filters.Loading || Contacts == null)
+        {
+            <div class="row">
+                <div class="col-12 alert alert-info">
+                    Loading...
+                </div>
+            </div>
+        }
+        @if (Contacts != null && Contacts.Count == 0)
+        {
+            <div class="row">
+                <div class="col-12 alert alert-warning">
+                    No contacts found.
+                </div>
+            </div>
+        }
+        @if (Contacts != null)
+        {
+            @foreach (var contact in Contacts)
+            {
+                <ContactRow CurrentContact="@contact"
```

Comment:
It's very valuable to use `@key` when rendering components inside a loop and the set can change over time. I'd recommend that here.

Reply:
Great suggestion. Done.